Plutarch/Builtin/List/Type.hs

```diff
@@ -0,0 +1,48 @@
+module Plutarch.Builtin.List.Type (
+  PList (..),
+  ListElemUni (..),
+) where
+
+import qualified Data.ByteString as BS
+import Data.Proxy (Proxy (..))
+import Data.Text (Text)
+import Plutarch (POpaque, Term)
+import Plutarch.Bool (PBool)
+import Plutarch.ByteString (PByteString)
+import Plutarch.Integer (PInteger)
+import Plutarch.Prelude (Type)
+import Plutarch.String (PString)
+import Plutarch.Unit (PUnit)
+import qualified PlutusCore as PLC
+import qualified PlutusCore.Data as PLC
+
+data PList a s
```

Comment:
Please don't move just the data type definition into a separate file. This means you make orphan instances, in addition you generally always want to use both modules in tandem, never just one.